gratipay/models/participant/email.py

```diff
@@ -201,63 +202,107 @@ def start_package_claims(self, c, nonce, *packages):
                                )
 
 
-    def update_email(self, email):
-        """Set the email address for the participant.
+    def set_primary_email(self, email, cursor=None):
+        """Set the primary email address for the participant.
         """
-        if not getattr(self.get_email(email), 'verified', False):
-            raise EmailNotVerified()
-        username = self.username
-        with self.db.get_cursor() as c:
-            self.app.add_event( c
-                              , 'participant'
-                              , dict(id=self.id, action='set', values=dict(primary_email=email))
-                               )
-            c.run("""
-                UPDATE participants
-                   SET email_address=%(email)s
-                 WHERE username=%(username)s
-            """, locals())
+        if cursor:
+            self._set_primary_email(email, cursor)
+        else:
+            with self.db.get_cursor() as cursor:
```

Comment:
Yeah, this is a bit of a goofy construction. https://github.com/gratipay/postgres.py/issues/40 would help.